hazelcast-jet-core/src/main/java/com/hazelcast/jet/Observable.java

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet;
+
+import com.hazelcast.jet.function.Observer;
+import com.hazelcast.jet.impl.observer.BlockingIteratorObserver;
+import com.hazelcast.jet.pipeline.Sinks;
+import com.hazelcast.ringbuffer.Ringbuffer;
+
+import javax.annotation.Nonnull;
+import java.util.Iterator;
+import java.util.Objects;
+import java.util.Spliterator;
+import java.util.Spliterators;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Function;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+/**
+ * Represents a flowing sequence of events produced an
+ * {@link Sinks#observable(String) observable sinks}. To observe the events,
+ * call {@link #addObserver jet.getObservable(name).addObserver(myObserver)}.
+ * <p>
+ * The {@code Observable} is backed by a {@link Ringbuffer}, which, once
+ * created, has a fixed capacity for storing messages. It supports reading
+ * by multiple {@code Observer}s, which will all observe the same sequence of
+ * messages. A new {@code Observers} will start reading automatically from the
+ * oldest sequence available. Once the capacity is full, the oldest messages
+ * will be overwritten as new ones arrive.
+ * <p>
+ * The {@code Ringbuffer}'s capacity defaults to 10,000, but can be changed
```

Comment:
how do we change the default capacity?

Reply:
I don't think we can right now. It's a value hardcoded in code, see `RingbufferConfig.DEFAULT_CAPACITY` and how `RingbufferConfig.capacity` gets initialised with it at instantiation.